tfhe/src/shortint/server_key/mod.rs

```diff
@@ -913,6 +922,14 @@ impl ServerKey {
         ct: &Ciphertext,
         acc: &ManyLookupTableOwned,
     ) -> Vec<Ciphertext> {
+        if cfg!(any(feature = "noise-asserts", test)) && ct.noise_level() != NoiseLevel::UNKNOWN {
```

Comment:
`NoiseLevel::UNKNOWN` and `NoiseLevel::MAX` have the same value which means a `NoiseLevel::MAX` would not be checked here
It seems to me that it would be better to panic if we have `NoiseLevel::MAX` here

Reply:
I think it would be better to make the values of unknown and max different, because I do need the distinction because of some things do in the casting keys

Reply:
The idea is that if the noise level is unknown we may allow to do the PBS, but not if'ts at MAX

In the casting keys, at some point, there's a ciphertext that is set with noise_level unknown and we do a PBS on it, so either we allow to do PBS on NoiseLevel::UNKNOWN noise level, or We change the thing in the casting key to not be at NoiseLevel::UNKNOWN

Reply:
so here to note that the MAX is usize::MAX i.e. an impossible value to reach, UNKNOWN was set to MAX as a safety

Reply:
maybe we need to think about using

MAX = usize::MAX - 1
UNKNOWN = usize::MAX

and see if that could make sense in the general case

given that a noise of usize::MAX - 1

would break anything

also maybe time to change the noise level stuff to use u32 or u64 instead of usize, given that those don't refer to memory 

not necessarily in this PR but it begs the question